examples/embeddable_examples/public/react_embeddables/saved_book/book_state.ts

```diff
@@ -7,40 +7,20 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { BehaviorSubject } from 'rxjs';
+import { WithAllKeys } from '@kbn/presentation-publishing';
 import { BookAttributes, BookAttributesManager } from './types';
 
-export const defaultBookAttributes: BookAttributes = {
+export const defaultBookAttributes: WithAllKeys<BookAttributes> = {
   bookTitle: 'Pillars of the earth',
   authorName: 'Ken follett',
   numberOfPages: 973,
   bookSynopsis:
     'A spellbinding epic set in 12th-century England, The Pillars of the Earth tells the story of the struggle to build the greatest Gothic cathedral the world has known.',
 };
 
-export const stateManagerFromAttributes = (attributes: BookAttributes): BookAttributesManager => {
-  const bookTitle = new BehaviorSubject<string>(attributes.bookTitle);
-  const authorName = new BehaviorSubject<string>(attributes.authorName);
-  const numberOfPages = new BehaviorSubject<number>(attributes.numberOfPages);
-  const bookSynopsis = new BehaviorSubject<string | undefined>(attributes.bookSynopsis);
-
-  return {
-    bookTitle,
-    authorName,
-    numberOfPages,
-    bookSynopsis,
-    comparators: {
-      bookTitle: [bookTitle, (val) => bookTitle.next(val)],
-      authorName: [authorName, (val) => authorName.next(val)],
-      numberOfPages: [numberOfPages, (val) => numberOfPages.next(val)],
-      bookSynopsis: [bookSynopsis, (val) => bookSynopsis.next(val)],
-    },
-  };
-};
-
 export const serializeBookAttributes = (stateManager: BookAttributesManager): BookAttributes => ({
```

Comment:
You can remove `serializeBookAttributes` method since you can just call `stateManager.getLatestState()` instead